pkg/antctl/raw/upgrade/apistorage/command.go

```diff
@@ -0,0 +1,261 @@
+		return err
+	}
+
+	crdObjectsToUpgrade := getCRDListToUpgrade(cmd, opts.k8sClient, crdNameListToUpgrade)
+	for _, crd := range crdObjectsToUpgrade {
+		if err = upgradeCRDObjects(cmd, opts.k8sClient, crd); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to upgrade CRD %q: %v.\n", crd.Name, err)
+		}
+		if err = patchCRDStoredVersions(opts.k8sClient, crd); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to patch CRD %q status.storedVersion: %v.\n", crd.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// getAllAntreaCRDNames is used to get all CRD names whose suffix is "crd.antrea.io".
+func getAllAntreaCRDNames(k8sClient client.Client) (sets.Set[string], error) {
+	crdNameList := sets.New[string]()
+	crdList := &apiextv1.CustomResourceDefinitionList{}
+	if err := k8sClient.List(context.TODO(), crdList); err != nil {
+		return nil, fmt.Errorf("failed to get CRD list: %v", err)
+	}
+	for _, crd := range crdList.Items {
+		if strings.HasSuffix(crd.Name, "crd.antrea.io") {
+			crdNameList.Insert(crd.Name)
+		}
+	}
+
+	return crdNameList, nil
+}
+
+// getCRDNameListToUpgrade is used to get the name list of CRDs to upgrade.
+func getCRDNameListToUpgrade(cmd *cobra.Command, k8sClient client.Client, crdNameListToUpgrade sets.Set[string]) (sets.Set[string], error) {
+	allCRDNameList, err := getAllAntreaCRDNames(k8sClient)
+	if err != nil {
+		return nil, err
+	}
+	// If the user-provided list of CRDs to upgrade is empty, upgrade all CRDs managed by Antrea.
+	if crdNameListToUpgrade.Len() == 0 {
+		crdNameListToUpgrade = allCRDNameList
+	} else {
+		// If the user-provided list of CRDs to upgrade is not empty, and it contains CRDs which are managed by Antrea,
+		// skip the CRDs which are not managed by Antrea.
+		for name := range crdNameListToUpgrade.Difference(allCRDNameList) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Skip CRD %q which is not created by Antrea.\n", name)
+		}
+		// Only upgrade the CRDs managed by Antrea.
+		crdNameListToUpgrade = crdNameListToUpgrade.Intersection(allCRDNameList)
+	}
+	return crdNameListToUpgrade, nil
+}
+
+// getCRDListToUpgrade is used to get the list of CRDs to upgrade.
+func getCRDListToUpgrade(cmd *cobra.Command, k8sClient client.Client, crdNameListToUpgrade sets.Set[string]) []*apiextv1.CustomResourceDefinition {
+	var crdsToUpgrade []*apiextv1.CustomResourceDefinition
+	for crdName := range crdNameListToUpgrade {
+		crd := &apiextv1.CustomResourceDefinition{}
+		if err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: crdName}, crd); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to get CRD %q: %v.\n", crdName, err)
+			continue
+		}
+		// The CRD that has only one version does not need to be upgraded.
+		if len(crd.Spec.Versions) < 2 {
+			fmt.Fprintf(cmd.OutOrStdout(), "Skip upgrading CRD %q since it only has one version.\n", crdName)
+			continue
+		}
+		crdsToUpgrade = append(crdsToUpgrade, crd)
+	}
+	return crdsToUpgrade
+}
+
+// upgradeCRDObjects is used to upgrade the objects of a CRD.
```

Comment:
You used "resources", not "objects" in documentation. Use the same term.

Reply:
Term "objects" is used.